### PR TITLE
Prevent excessive caching of HTML files

### PR DESCRIPTION
**Pull request by bnlb**
@andryanmiller noticed that the Cache-Control header was set with a max age of 1 year for all requests to `/developers/*`. We only want a 1 year max for requests for assets, not for HTML.

This is working for me locally if I comment out the dev proxy code here:

```
if (!isProduction) {
  const devCenterProxyConfig = {
    target: config.env.stage.devCenterUrl,
    changeOrigin: true
  }
  app.use(require('http-proxy-middleware')(devCenterProxyConfig))
}
```

I'll verify this in test as well before tomorrow's deploy.

### CODE REVIEW

File: server/utils/isHtmlRequest.spec.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      'https://algorithmia.com/developers/getting-started.html?foo=bar#fragment',
+    ].forEach(path => {
+      expect(isHtmlRequest(path)).to.be.true
+    })
+  })
+
+  it('should return false for all other requests', () => {
+    [
+      '',
+      'developers/',
```
Comment (langlk) [inline server/utils/isHtmlRequest.spec.js]: Okay, so we still want the cache header set to a year on things like `/developers/`?

Comment (bnlb) [reply server/utils/isHtmlRequest.spec.js]: Technically this won't happen, as a request to `/developers` will be updated to a request to `/developers/index.html` behind the scenes. This is more just for the sake of testing my regex.

Comment (langlk) [reply server/utils/isHtmlRequest.spec.js]: Ah, I guess we add the `.html` to these before we check for it in the server file.

**Comment by bnlb** [comment]
> LGTM–ultimately, this is one difficulty of needing to serve _all_ static content under `/developers`. I don't remember the ticket # that has us removing the hard-dependency on everything going under `/developers`, but I'd say let's comment on that ticket (or create a new one) that has us breaking out static assets into a separate dir that we can apply a specific cache-control header to.

https://algorithmia.atlassian.net/browse/FRONTEND-1940

**Pull request merged**